### PR TITLE
feat: Creating Executables Through Docker

### PR DESCRIPTION
**Pull request by AndreiUngur**
## Description
Builds on top of https://github.com/scrum-gang/neutralinojs/pull/15

Successfully creates executables with Docker + CMAKE

## Changes proposed
Add various missing libraries & dependencies to the Make build.

## How to test it

 - `docker build . -t test`
 - `docker run -it test`

## Next steps
Verify cross-platform capabilities

## Deploy notes

None.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -7,6 +7,19 @@ set(CMAKE_CXX_EXTENSIONS OFF)
 
 set(LINK_LIBS "")
 
+# Use the package PkgConfig to detect GTK+ headers/library files
```
Comment (kraglalbert) [inline CMakeLists.txt:10]: Make sure to rebase on the latest master, I believe Erick moved this into the Linux-specific block

File: Dockerfile [hunk 2]
```diff
@@ -1,16 +1,23 @@
-FROM andreiungur/ecse437:cmake-base
+FROM andreiungur/ecse437:cmake-base2
 
 WORKDIR /neutralinojs
 COPY . /neutralinojs
-CMD cmake .
+CMD cd build && cmake -DCMAKE_C_COMPILER=/usr/bin/gcc-8 -DCMAKE_CXX_COMPILER=/usr/bin/g++-8 /neutralinojs && make
 
 # Sets up base docker image
```
Comment (kraglalbert) [inline Dockerfile]: Maybe update this comment to say that the commands below are what's included in the `andreiungur/ecse437:cmake-base2` base image

**Review by kraglalbert — COMMENTED** [review]
Left some minor comments, I'll approve once this gets rebased onto master

File: CMakeLists.txt (hunk 1 above)
Comment (AndreiUngur) [reply CMakeLists.txt:10]: Erick's PR is not merged yet (https://github.com/scrum-gang/neutralinojs/pull/17) as it depends on mines!

File: Dockerfile (hunk 2 above)
Comment (AndreiUngur) [reply Dockerfile]: Good call, done!

**Review by kraglalbert — APPROVED** [review]
LGTM!

**Review by erickzhao — APPROVED** [review]
WFM on #17

**Pull request merged**